sretoolbox/utils/cache.py

```diff
@@ -0,0 +1,167 @@
+"""
+Caching helper decorator
+"""
+
+import functools
+import weakref
+
+_CACHES = weakref.WeakKeyDictionary()
+
+
+def static(key):
+    """
+    Decorates a class method to look up and cache the execution result.
+
+    Cache access is scoped by 2 things: class instance and `key`
+
+    :param key: static cache key
+    :type key: string
+    """
+    def decorate(func):
+        @functools.wraps(func)
+        def with_cache_get(self, *args, **kwargs):
+            cache = _lazy_init(self)
+            value = None
+            try:
+                value = cache[key]
+            except KeyError:
+                value = func(self, *args, **kwargs)
+                cache[key] = value
+            return value
+        return with_cache_get
+    return decorate
+
+
+def computed(key_func):
```

Comment:
the computed variant is giving users even more autonomy over the keys they use... I'd go in the oposite direction, but if you can convince me we need users providing the keys, then this function might come handy.